packages/valory/skills/decision_maker_abci/behaviours/purchase_subscription.py

```diff
@@ -0,0 +1,233 @@
+        # and the purchase tx: https://gnosisscan.io/tx/0x086e693fac1c3d084dffdc899adaca773ad986e6e2bff5700549e1ae8364731e (this is what the service should do)
+
+        purchase_params = {}
+
+        mech = self.params.mech_agent_address
+        params = self.params.mech_to_subscription_params[mech]
+
+        base_url = params["base_url"]
+        did = params["did"]
+        purchase_params["did"] = did
+        purchase_params["nft_receiver"] = self.synchronized_data.safe_contract_address
+        purchase_params["contract_address"] = params["transfer_nft_condition_address"]
+
+        did_url = f"{base_url}/{did}"
+        response = yield from self.get_http_response(
+            method="GET",
+            url=did_url,
+            headers={"accept": "application/json"},
+        )
+        if response.status_code != 200:
+            self.context.logger.error(
+                f"Could not retrieve data from did url {did_url}. "
+                f"Received status code {response.status_code}."
+            )
+            return None
+
+        try:
+            # TODO: would be good to get a sanity check here, whether its possible to get this information from somewhere else
+            data = json.loads(response.body)["data"]
+            service = data["service"]
+            for s in service:
+                if s["type"] == "nft_sales":
+                    template = s["attributes"]["serviceAgreementTemplate"]
+                    conditions = template["conditions"]
+                    for condition in conditions:
+                        if condition["type"] == "transferNFT":
+                            transfer_params = condition["parameters"]
+                            for param in transfer_params:
+                                if param["name"] == "_nftHolder":
+                                    purchase_params["nft_holder"] = param["value"]
+                                elif param["name"] == "_numberNfts":
+                                    purchase_params["nft_amount"] = int(param["value"])
+                                elif param["name"] == "_contractAddress":
+                                    purchase_params[
+                                        "nft_contract_address"
+                                    ] = param["value"]
+
+        except (ValueError, TypeError) as e:
+            self.context.logger.error(
+                f"Could not parse response from nervermined api, "
+                f"the following error was encountered {type(e).__name__}: {e}"
+            )
+            return None
+
+        # TODO
+        # purchase_params["agreement_id"] = "agreement_id"
+        # purchase_params["lock_payment_condition"] = "lock_payment_condition"
+        # purchase_params["transfer"] = "transfer", this might be always False from the txs I analyzed
+        # purchase_params["expiration_block"], this might be always 0 from the txs I analyzed
```

Comment:
`0` mints the NFT subscription doesn't expire. That parameter is the most common when dealing with NFT subscriptions limited by credits and not time limited (number of blocks)